docker/docker_image_src.go

```diff
@@ -39,6 +40,15 @@ func newImageSource(ctx *types.SystemContext, ref dockerReference, requestedMani
 	if requestedManifestMIMETypes == nil {
 		requestedManifestMIMETypes = manifest.DefaultRequestedManifestMIMETypes
 	}
+	// if people has requested an OCI manifest here but the docker registry doesn't
+	// still understand it we should force add a request for docker v2s2. Otherwise
+	// the registry won't understand an OCI manifest request and fallback to v2s1
+	// and OCI doesn't understand it (it cannot convert from v2s1 to OCI)
+	for _, mt := range requestedManifestMIMETypes {
+		if mt == imgspecv1.MediaTypeImageManifest {
+			requestedManifestMIMETypes = append(requestedManifestMIMETypes, manifest.DockerV2Schema2MediaType)
```

Comment:
This is ugly and suprising for an `ImageSource` to do (consider a caller who does not expect to use `image.Image`). Can we instead add the v2s1→OCI conversion? It might well be possible to do by creating a temporary v2s2 `memoryImage` (using the existing code) and then using the existing v2s2→OCI conversion.